src/viperleed/calc/classes/state_recorder.py

```diff
@@ -0,0 +1,49 @@
+"""Module state_recorder of viperleed.calc.classes.
+
+Defines a class to record and retrieve calculation states.
+"""
+
+__authors__ = (
+    'Alexander M. Imre (@amimre)',
+    )
+__copyright__ = 'Copyright (c) 2019-2024 ViPErLEED developers'
+__created__ = '2024-08-06'
+__license__ = 'GPLv3+'
+
+from collections import namedtuple
+from copy import deepcopy
+
+from viperleed.calc.sections.calc_section import CalcSection
+
+# A state is given by a tuple (slab, rparams).
+State = namedtuple('State', 'slab, rparams, section')
+
+class StateRecorder:
+    """Class that records and retrieves calculation states."""
+
+    def __init__(self):
+        """Initialize the state recorder with an empty list of states."""
+        self.recorded_states = []
```

Comment:
You may want to make this an underscore attribute (`_recorded_states`) given that you do give access to it via getter methods. Otherwise we risk that people (i.e., ourselves of the future) will start adding wrong stuff to it.